### PR TITLE
feat: Add Github Actions support

### PR DESCRIPTION
**Pull request by Hawkurane**
Fixes https://github.com/jhipster/jhipster-lite/issues/422

You can do:

POST /api/projects/init
POST /api/github-actions/init

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/config/GithubActionsBeanConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package tech.jhipster.lite.generator.githubactions.infrastructure.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.objenesis.SpringObjenesis;
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/config/GithubActionsBeanConfiguration.java:5]: Strange import

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  private final GithubActionsApplicationService githubActionsApplicationService;
+
+  GithubActionsResource(GithubActionsApplicationService githubActionsApplicationService) {
+    this.githubActionsApplicationService = githubActionsApplicationService;
+  }
+
+  @Operation(summary = "Init Github Actions YML file")
+  @ApiResponse(responseCode = "500", description = "An error occurred while initializing the github-actions.yml.mustache file")
+  @PostMapping("/init")
+  @GeneratorStep(id = "init")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:29]: Should be github-actions, as init is already used

File: src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsDomainService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  @Override
+  public void init(Project project) {
+    addYml(project);
+  }
+
+  @Override
+  public void addYml(Project project) {
+    projectRepository.template(project, SOURCE, GITHUB_ACTIONS_YML, GITHUB_ACTIONS_FOLDER);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsDomainService.java:25]: You forgot to add setup config
See https://github.com/jhipster/jhipster-lite/blob/main/.github/actions/setup/action.yml

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
I'm changing it to draft, as there are still some work to do

**Comment by Hawkurane** [comment]
Hello @pascalgrimaud i've just pushed the changes concerning the reviews you've made me.

Concerning the generate.sh i'm unsure how to approach this since I don't think there's any real way of checking generated github-actions files besides visually looking the website itself...

If you have any advice LMK :)

**Comment by pascalgrimaud** [comment]
Thanks a lot @Hawkurane 

About real final tests, I plan to automate a generated project + push to a repository, so it will trigger github actions there :-)

File: src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsService.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package tech.jhipster.lite.generator.githubactions.domain;
+
+import tech.jhipster.lite.generator.project.domain.Project;
+
+public interface GithubActionsService {
+  void init(Project project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsService.java:6]: I'd rename to: `addGitHubActionsForMaven`

File: src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsService.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+  void addYmls(Project project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/domain/GithubActionsService.java:8]: not sure we need this

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import tech.jhipster.lite.generator.githubactions.application.GithubActionsApplicationService;
+import tech.jhipster.lite.generator.project.domain.Project;
+import tech.jhipster.lite.generator.project.infrastructure.primary.dto.ProjectDTO;
+import tech.jhipster.lite.technical.infrastructure.primary.annotation.GeneratorStep;
+
+@RestController
+@RequestMapping("/api/github-actions/")
+@Tag(name = "Github Actions")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:17]:
```suggestion
@Tag(name = "GitHub Actions")
```

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+class GithubActionsResource {
+
+  private final GithubActionsApplicationService githubActionsApplicationService;
+
+  GithubActionsResource(GithubActionsApplicationService githubActionsApplicationService) {
+    this.githubActionsApplicationService = githubActionsApplicationService;
+  }
+
+  @Operation(summary = "Init Github Actions YML file")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:26]:
```suggestion
  @Operation(summary = "Add GitHub Actions YML files")
```

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  @ApiResponse(responseCode = "500", description = "An error occurred while initializing the github-actions.yml.mustache file")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:27]:
```suggestion
  @ApiResponse(responseCode = "500", description = "An error occurred while adding GitHub Actions")
```

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  @PostMapping("/init")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:28]:
```suggestion
  @PostMapping("/maven")
```

File: src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  @GeneratorStep(id = "github-actions")
+  public void init(@RequestBody ProjectDTO projectDTO) {
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/githubactions/infrastructure/primary/rest/GithubActionsResource.java:30]: I'd rename to `addGitHubActionsForMaven`, but if you don't like it, simply ignore my comment

File: src/main/resources/generator/githubactions/.github/workflows/github-actions.yml.mustache [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+      uses: actions/cache@v2
+      with:
+      path: ~/.m2/repository
+      key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+      restore-keys: |
+        ${{ runner.os }}-maven-<%={{ }}=%>
+    - name: 'Test: run tests'
+      run: |
+        chmod +x mvnw
+        ./mvnw clean verify
```
Comment (pascalgrimaud) [inline src/main/resources/generator/githubactions/.github/workflows/github-actions.yml.mustache:27]: Problem with indent

```suggestion
    runs-on: ubuntu-latest
    timeout-minutes: 20
    steps:{{=<% %>=}}
      - name: 'Setup: checkout project'
        uses: actions/checkout@v2
      - name: 'Setup: environment'
        id: setup
        uses: ./.github/actions/setup
      - name: 'Init: cache local Maven repository'
        uses: actions/cache@v2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
      - name: 'Test: run tests'
        run: |
          chmod +x mvnw
          ./mvnw clean verify
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
and your code works well (after fixing the indent): see https://github.com/pascalgrimaud/jhlite-sample-app/pull/1

**Comment by Hawkurane** [comment]
@pascalgrimaud thank you for the review,
In the end, i've added a callApi "/api/github-actions/maven" step every time there's a call for the maven build-tools. WDYT?

**Comment by pascalgrimaud** [comment]
@Hawkurane : I'd prefer to put it in `fullapp` as it's specific, but in the end, it won't impact the build, so it's fine, like you wish :)

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**